internal/proxy/task_scheduler.go

```diff
@@ -170,6 +172,24 @@ func (queue *baseTaskQueue) Enqueue(t task) error {
 		return err
 	}
 
+	if st, ok := t.(*searchTask); ok {
+		if skip, _ := st.CanSkipAllocTimestamp(); skip {
+			physicalTs := time.Now().UnixMilli()
+			ts := tsoutil.ComposeTS(physicalTs, rand.Int63n(physicalTs))
+			t.SetID(UniqueID(ts))
+			return queue.addUnissuedTask(t)
+		}
+	}
+
+	if qt, ok := t.(*queryTask); ok {
+		if skip, _ := qt.CanSkipAllocTimestamp(); skip {
+			physicalTs := time.Now().UnixMilli()
+			ts := tsoutil.ComposeTS(physicalTs, rand.Int63n(physicalTs))
+			t.SetID(UniqueID(ts))
+			return queue.addUnissuedTask(t)
+		}
+	}
+
```

Comment:
The logic is a little bit duplicate with below `SetID` & `addUnissuedTask`. Maybe we can consider only spliting how we allocate the timestamp.